src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -926,7 +943,11 @@ private void BuildSymbolsPackage(string path)
 
         internal void AnalyzePackage(PackageArchiveReader package)
         {
-            IEnumerable<IPackageRule> packageRules = Rules;
+            AnalyzePackage(package, Rules);
+        }
+
+        internal void AnalyzePackage(PackageArchiveReader package, IEnumerable<IPackageRule> packageRules)
```

Comment:
Something feels odd about this overload. Since you always want to use the Rules property, why not remove all the `Rules.Concat` calls, and instead make the `AnalyzePackage(PackageArchiveReader package, IEnumerable<IPackageRule> packageRules)` do the Concat once?

My thought is, either make `AnalyzePackage` depend on `this.Rules` (as I suggest above and is my preference), with the overload concatenating additional rules..... **OR** make `AnalyzePackage` always use a parameter, which you pass in this.Rules and possibly concat other collections as necessary.

Reply:
I suppose if the intent is for `AnalyzePackage` to be used for ad-hoc validation, then it can remain as is. But my impression is it should always search the Rules collection in its entirety, plus sometimes additional rules.

Reply:
I added `AnalyzePackage(PackageArchiveReader, IEnumerable<IPackageRule>)` to add the Icon deprecation message with custom messages depending on the case.

Reply:
I will revert back to the original method signature.